### PR TITLE
Bug fix: IndexNotFound error when API calls in environment where no Search Applications Exist

### PR DESCRIPTION
**Pull request by kderusso**
Captures IndexNotFound exceptions to return a more consistent message.

### CODE REVIEW

**Review by jimczi — APPROVED** [review]
Thanks @kderusso! Code LGTM, can you also add a yml test for this case?

**Review by carlosdelest — APPROVED** [review]
LGTM, just added a PR with some alternative for dealing with the onFailure() common code. 👍

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java [hunk 1]
```diff
@@ -184,15 +186,27 @@ private static XContentBuilder getIndexMappings() {
...
+                final SearchApplication res = parseSearchApplicationBinaryFromSource(source);
+                listener.onResponse(res);
             }
-            final BytesReference source = getResponse.getSourceInternal();
-            final SearchApplication res = parseSearchApplicationBinaryFromSource(source);
-            delegate.onResponse(res);
-        }));
+
+            @Override
+            public void onFailure(Exception e) {
```
[the remark this answers is not in the document]
Comment (kderusso) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java]: Nice optimizations, thank you!

**Comment by kderusso** [comment]
@elasticmachine run docs build

**Pull request merged**